dependencies/llvm/CMakeLists.txt

```diff
@@ -68,73 +68,98 @@ set(LLVM_COMPONENTS mcjit bitwriter linker passes)
 option(TARGET_AARCH64 "Include AARCH64 (arm64) target" ${WITH_AARCH64})
 if (TARGET_AARCH64)
     target_compile_definitions(Halide_LLVM INTERFACE WITH_AARCH64)
-    list(APPEND LLVM_COMPONENTS AArch64)
+    if (NOT LLVM_USE_SHARED_LLVM_LIBRARY)
+        list(APPEND LLVM_COMPONENTS AArch64)
+    endif (NOT LLVM_USE_SHARED_LLVM_LIBRARY)
```

Comment:
`endif` hasn't required a duplicated condition since CMake 2.x. We require 3.16+, so this isn't getting checked in like that.

But it would be better to drop this and the similar changes below it and just overwrite `LLVM_COMPONENTS` as I suggest below.